app/models/mixins/provider_proxy_mixin.rb

```diff
@@ -0,0 +1,13 @@
+module ProviderProxyMixin
+  extend ActiveSupport::Concern
+
+  def http_proxy_uri
+    proxy = VMDB::Util.http_proxy_uri("#{emstype}_http_proxy".to_sym)
```

Comment:
Or something like the refresher:
https://github.com/durandom/manageiq/blob/c445fb3fd2b40e3e2a3377e2ed78bb20f12589ed/app/models/manageiq/providers/base_manager/refresher.rb#L22